components/oauth-extensions/dynamic-client-manager/src/main/java/org/wso2/carbon/identity/oauth/extension/impl/ClientRegistrationServiceImpl.java

```diff
@@ -134,7 +130,8 @@ private OAuthApplicationInfo registerApplication(RegistrationProfile profile) th
     }
 
     public OAuthApplicationInfo createOAuthApplication(
-            String userId, String applicationName, String callbackUrl) throws APIManagementException, IdentityException {
+            String userId, String applicationName, String callbackUrl, String grantType)
```

Comment:
Check the possibility of using a bean to wrap all these input parameters. Ideally, you should be able to re-use "RegistrationProfile" bean to do this.